README.md

```diff
@@ -15,6 +15,27 @@ for a general introduction on how to use this crate, and
 [API docs](https://jorgecarleitao.github.io/arrow2/main/docs/arrow2)
 for a detailed documentation of each of its APIs.
 
+# 2023-04-12: Deprecation Notice
```

Comment:
```suggestion
# 2023-04-12: Maintenance Notice
```

Maybe if we change the title so it didn't say "deprecated" this would sound less scary 🤔